src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInMetricsListener.cs

```diff
@@ -82,20 +82,44 @@ public override void OnEventWritten(string name, object payload)
 
                 TagList tags = default;
 
-                tags.Add(new KeyValuePair<string, object>(SemanticConventions.AttributeHttpFlavor, HttpTagHelper.GetFlavorTagValueFromProtocol(context.Request.Protocol)));
-                tags.Add(new KeyValuePair<string, object>(SemanticConventions.AttributeHttpScheme, context.Request.Scheme));
-                tags.Add(new KeyValuePair<string, object>(SemanticConventions.AttributeHttpMethod, context.Request.Method));
-                tags.Add(new KeyValuePair<string, object>(SemanticConventions.AttributeHttpStatusCode, TelemetryHelper.GetBoxedStatusCode(context.Response.StatusCode)));
+                // see the spec https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/trace/semantic_conventions/http.md
+                if (this.httpSemanticConvention.HasFlag(HttpSemanticConvention.Old))
```

Comment:
Some thoughts...

* Just reading this work I found "New" and "Old" pretty confusing. Why not `Legacy` vs `Stable` or something along those lines?

* This is probably an issue for some users: https://github.com/open-telemetry/opentelemetry-dotnet/blob/5d321ce621a387b4cff226ecc1dbcc5f3b6611bb/src/OpenTelemetry.Api/Internal/HttpSemanticConventionHelper.cs#L53

   We should be using `IConfiguration` to get the switch.

* Shouldn't the environment key have HTTP in the name like `OTEL_HTTP_SEMCONV_STABILITY_OPT_IN`?

Reply:
> * Just reading this work I found "New" and "Old" pretty confusing. Why not `Legacy` vs `Stable` or something along those lines?

+1, I had similar concern here https://github.com/open-telemetry/opentelemetry-dotnet/pull/4538#discussion_r1232783666

Reply:
1. > Just reading this work I found "New" and "Old" pretty confusing. Why not Legacy vs Stable or something along those lines?

How strongly do you feel about this? :)
We can change the enum, but `New` and `Old` shipped as part of OpenTelemetry.Api v1.5.0.
It's `internal` so we CAN change it.
But this would require re-releasing the Api library. (ie Instrumentation.AspNetCore would require an OpenTelemetry.Api v1.5.1).

Also, this is meant to be short lived. When these instrumentation libraries go stable this enum and helper class will be removed.

2. > We should be using IConfiguration to get the switch.

Can you please share an example of this?

3. > Shouldn't the environment key have HTTP in the name like OTEL_HTTP_SEMCONV_STABILITY_OPT_IN?

That makes sense to me, but the spec defines [OTEL_SEMCONV_STABILITY_OPT_IN](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#:~:text=OTEL_SEMCONV_STABILITY_OPT_IN)

Reply:
> We can change the enum, but New and Old shipped as part of OpenTelemetry.Api v1.5.0.
It's internal so we CAN change it.
But this would require re-releasing the Api library. (ie Instrumentation.AspNetCore would require an OpenTelemetry.Api v1.5.1).

The instrumentation libraries do not use this `enum` from `OpenTelemetry.Api`. They link to the file so each of the libraries have their own copy of it. You should be able to make the necessary changes without worrying about version of `OpenTelemetry.Api`.

Note: This does not have to be done in this PR.